### PR TITLE
Move the meat of getPlayableCards to canPlay(). This made some game instances superfluous.

### PR DESCRIPTION
**Pull request by kberg**
…nstances superfluous.

### CODE REVIEW

File: src/Player.ts [hunk 1]
```diff
@@ -1594,63 +1594,65 @@ export class Player implements ISerializable<SerializedPlayer> {
...
-      // reserve costs that a player cannot afford.
-      if (card.reserveUnits !== undefined) {
-        const reserveUnits = MoonExpansion.adjustedReserveCosts(this, card);
-        // If there isn't enough steel to meet the purchase reserve, this isn't a playable card.
-        if (steel < reserveUnits.steel) {
-          return false;
-        }
-        // Set aside reserve units in case the card has a building tag.
-        steel = Math.max(0, steel - reserveUnits.steel);
+    const playableCards = candidateCards.filter((card) => this.canPlay(card));
```
Comment (chosta) [inline src/Player.ts:1597]: If `playableCards` is not used elsewhere can't we just do:
`return candidateCards.filter((card) => this.canPlay(card))`

**Review by chosta — APPROVED** [review]

File: src/Player.ts [hunk 2]
```diff
@@ -1594,63 +1594,65 @@ export class Player implements ISerializable<SerializedPlayer> {
...
+    if (card.reserveUnits !== undefined) {
+      const reserveUnits = MoonExpansion.adjustedReserveCosts(this, card);
+      // If there isn't enough steel to meet the purchase reserve, this isn't a playable card.
+      if (steel < reserveUnits.steel) {
+        return false;
       }
-      if (canUseTitanium) {
-        maxPay += titanium * this.getTitaniumValue();
+      // Set aside reserve units in case the card has a building tag.
+      steel = Math.max(0, steel - reserveUnits.steel);
```
Comment (Lynesth) [inline src/Player.ts:1618]: This cannot be `< 0`. Same for titanium below.

Comment (kberg) [reply src/Player.ts:1618]: Most probably not. But I'd like to do some kind of defensive thing so someone don't wind up in a situation with a negative steel count. Or are you saying that we should remove that safety net?

Comment (Lynesth) [reply src/Player.ts:1618]: You have that net just above, that's what I mean.
```ts
      if (steel < reserveUnits.steel) {
        return false;
      }
```
So after that, you are sure that `steel - reserveUnits.steel` is `>= 0`.

Comment (kberg) [reply src/Player.ts:1620]: Oh! Yes. Thanks.

**Pull request merged**